### ISSUE TITLE
Some accounts and passphrase do not match

### ISSUE THREAD

**Issue by JoJa10** [Sneezry/authenticator#42]
Getting above reply when trying to enter passphrase.  I was able to open up dialog that lets me re-enter and now just see dots.

I found some of your suggestions.  I clicked on inspect element & selected console on menu.  Got the following, which is in red:
Error in response to storage.get: TypeError: Cannot read property 'length' of undefined
    at Object.jsSHA.getHMAC (chrome-extension://bhghoamapcdpbohphigoooaddinpkbai/totp/jssha-1.31.min.js:11:14553)
    at generate (chrome-extension://bhghoamapcdpbohphigoooaddinpkbai/totp/totp.js:71:22)
    at updateCode (chrome-extension://bhghoamapcdpbohphigoooaddinpkbai/javascript/popup.js:545:62)
    at Object.showCodes [as callback] (chrome-extension://bhghoamapcdpbohphigoooaddinpkbai/javascript/popup.js:678:9)
    at chrome-extension://bhghoamapcdpbohphigoooaddinpkbai/javascript/popup.js:66:21

**Comment by JoJa10** [comment]
I'm hoping I can get a response to the question above...

**Comment by Sneezry** [comment]
Hi @JoJa10 , sorry for late reply.

I do know some users meet the same issue as yours, however, I cannot reproduce on my own side :(

The log you provide means one of your secrets is illegal, a valid secret is in format of Base32.

And now, if you have backed up your secrets, unlock your accounts with backups, if unfortunately, not, you need contact with your service provider (such as Google, dropbox, etc) to help you to unlock your account.

To use this extension safely (I guess), please check "remember passphase".

**Comment by JoJa10** [comment]
Thank you for your reply.  I have some questions:
1. Wouldn't it be less safe to check 'remember passphrase'?  I thought it added extra security...
2. What do you mean by secrets - and how do I back them up?

**Comment by Sneezry** [comment]
Hi @JoJa10 , by saying safely, I meant you would not be locked again. Secrets is a string your service provider gives you when you enable 2FA. If you add 2FA by QR code, there should be a way to find out the secret in the 2FA setting page, and there should be a notice to mind you backup some code to make sure you will not be locked when your device does not work or is lost.

**Comment by JoJa10** [comment]
For the benefit of others: I logged into my account used for 2F & dis-associated the authenticator via link I found, then later had to re-associate, since this required for the org I work at.  Upon doing this, I got a new secret.  I managed to find the screen on your authenticator which allowed me to re-enter  the key - don't re-call what I did to get there, but I think it started with the edit icon.  After that,  the 6 digit number re-appeared!   I had also changed my passphrase via authenticator prior - not sure if that took or not & checked the 'remember passphrase'.
